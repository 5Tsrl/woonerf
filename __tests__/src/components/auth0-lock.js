// @flow
import React from 'react'
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import Auth0 from '../../../src/components/auth0-lock'

describe('auth0-lock', () => {
  it('should show lock and handle successful authentication', () => {
    window.localStorage = {
      setItem: () => null
    }

    const mockStore = configureStore()({})

    // mount component
    renderer.create(
      <Provider store={mockStore}>
        <Auth0
          lockOptions={{ fakeAuthenticatedToken: 'ok-token' }}
          />
      </Provider>
    )

    expect(mockStore.getActions()).toMatchSnapshot()
  })

  it('should show lock and handle unsuccessful authentication', () => {
    window.localStorage = {
      setItem: () => null
    }

    const mockStore = configureStore()({})

    // mount component
    renderer.create(
      <Provider store={mockStore}>
        <Auth0
          lockOptions={{ fakeAuthenticatedToken: 'bad-token' }}
          />
      </Provider>
    )

    expect(mockStore.getActions()).toMatchSnapshot()
  })
})
