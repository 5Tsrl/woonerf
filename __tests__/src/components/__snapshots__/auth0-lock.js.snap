// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth0-lock should show lock and handle successful authentication 1`] = `
Array [
  Object {
    "payload": Object {
      "idToken": "ok-token",
      "profile": Object {
        "profileData": "blah",
      },
    },
    "type": "set auth0 user",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;

exports[`auth0-lock should show lock and handle unsuccessful authentication 1`] = `
Array [
  Object {
    "payload": null,
    "type": "set auth0 user",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/login",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;
